config/config.go

```diff
@@ -474,6 +483,34 @@ func (c *ServersetSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	return checkOverflow(c.XXX, "serverset_sd_config")
 }
 
+// MarathonSDConfig is the configuration for services running on Marathon
+type MarathonSDConfig struct {
+	Name           string   `yaml:"name,omitempty"`
```

Comment:
I added the `name` field mostly for having unique source names for the target groups when you have more than one marathon cluster running (I construct the names like `marathon:<clustername>:<appname>`). Wouldn't having duplicate names break the updating of existing target groups?

The `UseJobPrefix` was added later as it seemed a good extension. There you are right, it could probably also be solved with relabeling.